CoreLib/SharedPtr.h

```diff
@@ -0,0 +1,490 @@
+﻿#pragma once
+
+
+#include "CoreLibConcepts.h"
+
+#include <Core.h>
+
+#include <stdint.h>
+
+namespace Fuego
+{
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    class SharedPtr;
+
+    template<class T, class Deleter = std::default_delete<T>> requires IsDefaultCompatibleDeleter<T, Deleter>
+    class FUEGO_API WeakPtr
+    {
+    public:
+
+        WeakPtr() noexcept = default;
+        WeakPtr(const WeakPtr<T, Deleter>& other) noexcept;
```

Comment:
What do you think about `using` for `WeakPtr<T, Deleter>` and `SharedPtr<T, Deleter>`?